nntrainer/tensor/tensor.h

```diff
@@ -841,8 +863,17 @@ class Tensor {
   TensorDim dim;
   std::array<unsigned int, MAXDIM> strides;
   bool is_contiguous;
+
   std::shared_ptr<float> data;
 
+  /**<
+   * When using shared_data with tensor, this stores the ptr of the source
+   * tensor which handles the full memory. If tensor data is already allocated,
+   * this does not affect the tensor. If the tensor data is not allocated, and
+   * src_ptr is valid, this tensor will use the memory allocated by the src_ptr
+   */
+  std::shared_ptr<SrcSharedTensor> src_tensor;
```

Comment:
I think, in the long term, we should have storage object or equivalent to manage the memory, tensor often used as a wrapper to a memory. Thus making our intentions more clear by using storage if we are to manage memory.

Reply:
Yes, the storage object would wrap both `data` and `src_tensor`.
It will be done. For now, this PR has been rushed to solved the bug ASAP.

Reply:
Yup